src/ripple/app/ledger/impl/LedgerCleaner.cpp

```diff
@@ -48,14 +41,24 @@ Cleans up the ledger. Specifically, resolves these issues:
 
 */
 
-class LedgerCleanerImp
-    : public LedgerCleaner
-    , public beast::Thread
+class LedgerCleanerImp : public LedgerCleaner
 {
-public:
     Application& app_;
-    beast::Journal m_journal;
-    std::mutex lock_;
+    beast::Journal j_;
+    mutable std::mutex mutex_;
+
+    mutable std::condition_variable wakeup_;
+
+    std::thread thread_;
+
+    enum class State : char {
+        notReady = 0,
+        startCleaning,
+        cleaning,
+        stoppedCleaning,
+        shouldExit
```

Comment:
Minor nitpick, `notReady` and `stoppedCleaning` are functionally equivalent and could be merged into one state, e.g. `readyToClean`